flink-java/src/main/java/org/apache/flink/api/java/io/CsvInputFormat.java

```diff
@@ -152,6 +177,38 @@ public void setFields(boolean[] sourceFieldMask, Class<?>[] fieldTypes) {
 	public Class<?>[] getFieldTypes() {
 		return super.getGenericFieldTypes();
 	}
+
+	public void setFieldsMap(String[] fieldsMap) {
+		Preconditions.checkNotNull(fieldsMap);
+		Preconditions.checkState(typeInformation instanceof PojoTypeInfo);
+
+		PojoTypeInfo<OUT> pojoTypeInfo = (PojoTypeInfo<OUT>) typeInformation;
+
+		String[] fields = pojoTypeInfo.getFieldNames();
+		Class<?>[] fieldTypes = getFieldTypes();
+		this.fieldsMap = Arrays.copyOfRange(fieldsMap, 0, fieldsMap.length);
+
+		boolean[] includeMask = new boolean[fieldsMap.length];
+		Class<?>[] newFieldTypes = new Class<?>[fieldsMap.length];
+
+		for (int i = 0; i < fieldsMap.length; i++) {
+			if (fieldsMap[i] == null) {
+				includeMask[i] = false;
+				newFieldTypes[i] = null;
+				continue;
+			}
+
+			for (int j = 0; j < fields.length; j++) {
+				if (fields[j].equals(fieldsMap[i])) {
+					includeMask[i] = true;
+					newFieldTypes[i] = fieldTypes[j];
+					break;
+				}
+			}
```

Comment:
Can you throw an exception if the provided field name was not found in the POJO type information?